packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -70,169 +96,258 @@ function WorklistDataGrid(props) {
+
+    if (hasSelectableRows && focusedRow.current === 0 && focusedCol.current === 0) {
+      // When row selection is turned on for the first time, default to the first row selection cell.
+      newFocusCell = { row: 1, col: 0 };
+    } else if (!hasSelectableRows && focusedCol.current === 0) {
+      // When row selection is turned off, if a cell in the row selection had focus, then
+      // refocus on the first cell in that row.
+      newFocusCell = { row: focusedRow.current, col: 0 };
+    }
+    setFocusedRowCol(newFocusCell.row, newFocusCell.col, false);
+    if (currentSelectedCell != null) {
+      setCurrentSelectedCell(null);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hasSelectableRows]);
 
-    focusedRow.current = nextRow;
-    focusedCol.current = nextCol;
+  const handleNavigationModeChange = (isGridNavigationEnabled) => {
+    ariaLiveMsg.current = intl.formatMessage({ id: isGridNavigationEnabled ? 'Terra.worklist-data-grid.navigation.enabled' : 'Terra.worklist-data-grid.navigation.disabled' });
 
-    const elementWithTabStop = grid.current.rows[nextRow].cells[nextCol];
-    elementWithTabStop.tabIndex = 0;
-    elementWithTabStop.focus();
-    event.preventDefault();
+    setIsNavigationEnabled(isGridNavigationEnabled);
   };
 
-  const handleClick = (event) => {
-    // Determine which cell was clicked. In the event that the user holds the mouse across multiple cells,
-    // the originating cell is the clicked cell/active element.
-    const clickedCell = event.target.closest('td,th') || document.activeElement.closest('td,th');
-    if (!clickedCell) {
-      // If anything other than a table data or table header cell is clicked, ignore the click.
-      return;
+  const areRowsSelected = () => (
+    rows.find(r => r.isSelected === true)
+  );
+
+  const clearRowSelection = () => {
+    ariaLiveMsg.current = '';
+    if (areRowsSelected()) {
+      ariaLiveMsg.current = intl.formatMessage({ id: 'Terra.worklist-data-grid.all-rows-unselected' });
+      // Esc (while in row selection mode and rows are selected): Clear selection
+      if (onClearSelectedRows) {
+        onClearSelectedRows();
+      }
+    } else if (onClearRowSelectionMode) {
+      ariaLiveMsg.current = intl.formatMessage({ id: 'Terra.worklist-data-grid.row-selection-mode-disabled' });
+      onClearRowSelectionMode();
+    }
+  };
+  const handleRowSelection = (rowId, selectAllRows, rowIndex) => {
+    // TODO: Consider changing the order of the parameter to (selectAllRows, rowId, rowIndex)
+    let msgId = 'Terra.worklist-data-grid.all-rows-selected';
+    if (!selectAllRows) {
+      // TODO: Find something to differentiate between row select and unselect. Row Id seems to be always valued.
+      msgId = rowId ? 'Terra.worklist-data-grid.row-selection-template' : 'Terra.worklist-data-grid.row-selection-unselected-template';
+    }
+    ariaLiveMsg.current = intl.formatMessage({ id: msgId }, { row: rowIndex });
```

Comment:
Rather than reading the row number, we should be reading a user provided name for the row that corresponds with the value in the row header.

Reply:
The row index will be the default identifier used in the select and unselect messages. If the user provides an ariaLabel for the row then that message will use that instead of the row index. If this needs to change, please document in Jira UXPLATFORM-9043.